SeqVision/modules/fastq_tool_module.py

```diff
@@ -0,0 +1,84 @@
+def calculate_gc_content(seq: str):
+    """
+    Calculates the content of G and C nucleotides in the sequence.
+
+    Parameters:
+    ----------
+    seq: str
+        String containing the nucleotide sequence of the read.
+
+    Returns:
+    ----------
+    int
+        Percentage of G and C nucleotides in the sequence.
+    """
+    gc_count = seq.upper().count("G") + seq.upper().count("C")
+    gc_content = gc_count / len(seq) * 100
+    return gc_content
+
+
+def filter_reads_by_gc(seq_line: str, gc_bounds=(0, 100)) -> bool:
+    """
+    Filters reads by GC content.
+
+    Parameters:
+    ----------
+    fastq_dict : dict
+        Dictionary where keys are read names and values are tuples
+        containing the sequence and quality.
+
+    gc_bounds : tuple or int or float, optional
+        GC content interval (in percent) for filtering.
+        If a single number is provided, it is considered the upper bound,
+        and the lower bound is set to 0.
+        Defaults to (0, 100), i.e., all reads are preserved.
+        Examples:
+        - (20, 80) - preserves only reads with GC content between 20% and 80%.
+        - 44.4 - preserves reads with GC content less than 44.4%.
+
+    Returns:
+    --------
+    dict
+        Filtered dictionary of reads
+        that meet the specified GC content conditions.
+    """
+    if isinstance(gc_bounds, (int, float)):
+        gc_bounds = (0, gc_bounds)
+    gc_content_count = calculate_gc_content(seq_line)
+    if gc_bounds[0] <= gc_content_count <= gc_bounds[1]:
+        return True
```

Comment:
Здесь и в функциях filter_length_bounds, filter_quality_threshold нет обработки случаев, когда условие фильтрации не выполнено -> ничего не вернется. Лучше в таком случае явно прописывать так, чтобы вернулся False

```suggestion
if условие:
   return True
return False
```